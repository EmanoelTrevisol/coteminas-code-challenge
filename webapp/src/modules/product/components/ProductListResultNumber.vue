<template>
  <h4>{{ resultString }}</h4>
</template>

<script>
export default {
  computed: {
    resultString() {
      const total = this.$store.state.product.total;
      if (total === 0) return 'nenhum produto encontrado';
      else if (total === 1) return '1 produto encontrado';
      else return `${total} produtos encontrados`;
    },
  },
};
</script>
