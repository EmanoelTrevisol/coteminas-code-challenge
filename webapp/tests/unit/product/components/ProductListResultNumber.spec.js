import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import ProductListResultNumber from '@/modules/product/components/ProductListResultNumber';

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    product: {
      namespaced: true,
      state: {
        filter: '',
        items: [],
        total: 0,
      },
      mutations: {
        // mocked function to test component filter update
        changeTotal(state, val) {
          state.total = val;
        },
      },
    },
  },
});

describe('ProductListResultNumber.vue', () => {
  const build = () => {
    const wrapper = shallowMount(ProductListResultNumber, {
      localVue,
      store,
    });

    return wrapper;
  };
  it('renders component', () => {
    const wrapper = build();

    expect(wrapper.text()).toMatchSnapshot();
  });

  it('no results found correct string', () => {
    const wrapper = build();

    const h4 = wrapper.find('h4');

    expect(h4.element.innerHTML).toBe('nenhum produto encontrado');
  });

  it('changes string correctly to plural when state changes to > 1 results', async () => {
    const wrapper = build();
    const total = 142;
    const newTitle = `${total} produtos encontrados`;

    store.commit('product/changeTotal', total);

    await localVue.nextTick();
    const h4 = wrapper.find('h4');

    expect(h4.element.innerHTML).toBe(newTitle);
  });

  it('changes string correctly with single result', async () => {
    const wrapper = build();
    const total = 1;
    const newTitle = `${total} produto encontrado`;

    store.commit('product/changeTotal', total);

    await localVue.nextTick();
    const h4 = wrapper.find('h4');

    expect(h4.element.innerHTML).toBe(newTitle);
  });
});
