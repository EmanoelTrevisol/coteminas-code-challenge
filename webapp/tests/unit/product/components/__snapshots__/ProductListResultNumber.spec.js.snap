// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListResultNumber.vue renders component 1`] = `"nenhum produto encontrado"`;
